adaptivealerting/detectors/builders/constant_threshold.py

```diff
@@ -0,0 +1,191 @@
+
+        if strategy == self.Strategy.SIGMA:
+            return self._create_sigma_detector(sample, weak_multiplier, strong_multiplier)
+        if strategy == self.Strategy.QUARTILE:
+            return self._create_quartile_detector(sample, weak_multiplier, strong_multiplier)
+        raise exceptions.DetectorBuilderError("Unknown build strategy")
+
+    def _create_sigma_detector(self, sample, weak_multiplier, strong_multiplier):
+        """Performs threshold calculations using sigma (standard deviation) strategy.
+
+        Parameters:
+            sample: list of number data on which calculations are performed
+            weak_multiplier: number that represents to multiplier to use to calculate weak
+                             thresholds
+            strong_multiplier: number that represents to multiplier to use to calculate strong
+                               thresholds
+
+        Returns:
+            Detector object
+       """
+
+        STRATEGY = self.Strategy.SIGMA
+
+        sigma = calculate_sigma(sample)
+        mean = calculate_mean(sample)
+        weak_upper_threshold, weak_lower_threshold = calculate_sigma_thresholds(
+            sigma, mean, weak_multiplier)
+        strong_upper_threshold, strong_lower_threshold = calculate_sigma_thresholds(
+            sigma, mean, strong_multiplier)
+
+        detector = ct.ConstantThresholdDetector(STRATEGY, weak_upper_threshold,
+                                                strong_upper_threshold, weak_lower_threshold,
+                                                strong_lower_threshold)
+        LOGGER.info("Detector created -- %s", detector)
+
+        return detector
+
+    def _create_quartile_detector(self, sample, weak_multiplier, strong_multiplier):
+        """Performs threshold calculations using quartile strategy.
+
+        Parameters:
+            sample: list of number data on which calculations are performed
+            weak_multiplier: number that represents to multiplier to use to calculate weak
+                             thresholds
+            strong_multiplier: number that represents to multiplier to use to calculate strong
+                               thresholds
+
+        Returns:
+            Detector object
+       """
+
+        STRATEGY = self.Strategy.QUARTILE
+
+        q1, median, q3 = calculate_quartiles(sample)
+        weak_upper_threshold, weak_lower_threshold = \
+                calculate_quartile_thresholds(q1, q3, weak_multiplier)
+        strong_upper_threshold, strong_lower_threshold = \
+                calculate_quartile_thresholds(q1, q3, strong_multiplier)
+
+        detector = ct.ConstantThresholdDetector(STRATEGY, weak_upper_threshold,
+                                                strong_upper_threshold, weak_lower_threshold,
+                                                strong_lower_threshold)
+        LOGGER.info("Detector created -- %s", detector)
+
+        return detector
+
+
+def calculate_sigma(sample):
```

Comment:
Shouldn't this be a "private" method too? (Similarly for the various calculate methods below.)

Reply:
Yes, you are correct. Fixed, thanks.